### DISCUSSION TITLE
WinForms, EnableSecurityPrincipalFlowFromClient and ApplicationContext.User null in the DataPortal

### DISCUSSION THREAD

**Discussion by richardbartley** [MarimerLLC/csla#3621] [Questions] [answered]
Its' 2023, but I have a WinForms NetCore project that is using Csla business library that we have happily already got an API using.
Now we need a WinForms client, and am authenticating against Azure AD using Microsofts MSAL library - ok so far.

Problem occurs when the Authorization rules fire as the Principal and User properties on ApplicationContext are null, so obviously the user is not authorized......

In 2023, with Csla 7.03 and a Winforms client, what's the setup needed?

As I understand from the docs, Windows authentication will NOT flow the identity to the server-side portal, so I've tried this config (I'm happy to accept the risk here) to force it.

```
services.AddCsla(options => options
    .AddWindowsForms()
    .DataPortal(dp =>
    {
        dp.EnableSecurityPrincipalFlowFromClient();
        dp.AuthenticationType("Csla");
    }));
```

My login code is successfully getting the user and their claims...and I'm setting ApplicationContext whereever I can (in hope :-))

```
var authResult = await AdLogin(); // Adlogin using MSAL library.
loginButton.Text = authResult.Account.Username;
...
var identity = new ClaimsIdentity(authResult.ClaimsPrincipal.Claims);
...
var cslaPrincipal = new Csla.Security.CslaClaimsPrincipal(identity);
Program.ApplicationContext.Principal = cslaPrincipal;
Thread.CurrentPrincipal = cslaPrincipal;
...
```

but my form blows up when trying to _portal.FetchAsync with the authentication rules throwing an exception as portal has not got any principal or user.

```
private async void ColourListPage_Load(object sender, EventArgs e)
{
    var list = await _portal.FetchAsync();
    colourListBox.DataSource = list.Select(p => p.ColourCode).ToList();
}
```

What am I doing wrong?
Thanks.

**Comment by rockfordlhotka** [comment, marked as answer]
The client and server need the same configuration. Have you configured the app server with `EnableSecurityPrincipalFlowFromClient`?
